index.js

```diff
@@ -72,56 +60,65 @@ function convert(resourceListing, apiDeclarations) {
       convertedSecurityNames);
   }
 
-  if (resourceListing.basePath) {
-    assignPathComponents(resourceListing.basePath, result);
-  }
+  var tagDescriptions = {};
+  resourceListing.apis.forEach(function(resource) {
+    var tagName = extractTag(resource.path);
+    if (!isValue(tagName)) { return; }
 
-  extend(models, resourceListing.models);
+    tagDescriptions[tagName] = resource.description;
+  });
 
   // Handle embedded documents
-  if (Array.isArray(resourceListing.apis)) {
-    if (apiDeclarations.length > 0) {
-      result.tags = [];
+  var resources = [resourceListing].concat(apiDeclarations);
+
+  resources.forEach(function(resource) {
+    var operationTags;
+    var tagName = extractTag(resource.resourcePath);
+
+    if (isValue(tagName)) {
+      result.tags = result.tags || [];
+      result.tags.push(extend({}, {
```

Comment:
In latter commit I will make wrapper function for `extend({}, ...)` trick with good describing name. 
Do you have suggestion for a name?